### PR TITLE
Servlets for agent beginning

### PR DESCRIPTION
**Pull request by Pragya1805**

### CODE REVIEW

File: FosServer/src/main/java/com/example/fostracker/servlets/AgentServlet/GetAgentServlet.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                String midName = resultSet.getString("AgentMidName");
+                if (midName.isEmpty()) {
+                    midName = "";
+                }
+                String lastName = resultSet.getString("AgentLastName");
+                agentEmail = resultSet.getString("AgentEmail");
+                String phone = resultSet.getString("AgentPhone");
+                Double latitude = resultSet.getDouble("AgentLatitude");
+                Double longitude = resultSet.getDouble("AgentLongitude");
+                agent = new Agent(new Name(firstName, midName, lastName), agentEmail, phone, new Coordinates(latitude, longitude));
```
Comment (trblessy) [inline FosServer/src/main/java/com/example/fostracker/servlets/AgentServlet/GetAgentServlet.java]: you can declare a variable for name and coordinates and then use that to create the agent object

File: FosServer/src/main/java/com/example/fostracker/servlets/AgentServlet/SpannerQueryFunctions.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    /**
+     * Makes spanner query for updating location of agent whose AgentEmail is email by setting
+     * AgentLatitude to latitude and AgentLongitude to longitude. Writes the number or rows
+     * affected in response writer.
+     *
+     * @param email
+     * @param latitude
+     * @param longitude
+     * @return
```
Comment (trblessy) [inline FosServer/src/main/java/com/example/fostracker/servlets/AgentServlet/SpannerQueryFunctions.java]: what is the return value? please add in the comment

File: FosServer/src/main/java/com/example/fostracker/servlets/AgentServlet/UpdateAgentLocation.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        Gson gson = new Gson();
+        EmailCoordinates emailCoordinates = gson.fromJson(jsonString, EmailCoordinates.class);
+
+        try {
+            long numberOfRowsChanged = SpannerQueryFunctions.updateAgentLocation(emailCoordinates.email, emailCoordinates.latitude, emailCoordinates.longitude);
+            response.setContentType("text");
+            response.setCharacterEncoding("UTF-8");
+            response.getWriter().write(Long.toString(numberOfRowsChanged));
+
+
```
Comment (trblessy) [inline FosServer/src/main/java/com/example/fostracker/servlets/AgentServlet/UpdateAgentLocation.java]: please remove extra newlines

File: FosServer/src/main/java/com/example/fostracker/servlets/FosTrackerServlet/GetNumberOfStoresByStatus.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                Long value = numberOfStoresByStatus.getLong("NumberOfStores");
+                String key = "UNVISITED";
+                if (numberOfStoresByStatus.isNull("VerificationStatus") == false){
+                    key = numberOfStoresByStatus.getString("VerificationStatus");
+                }
+                statusToNumberOfStoresMap.put(key, value);
+
+            } while (numberOfStoresByStatus.next());
+
+
```
Comment (trblessy) [inline FosServer/src/main/java/com/example/fostracker/servlets/FosTrackerServlet/GetNumberOfStoresByStatus.java]: Does formatter remove these whitespaces?

**Review by trblessy — CHANGES_REQUESTED** [review]
Overall a good PR. Try to the formatting tool before sending the PR.

File: FosServer/src/main/java/com/example/fostracker/servlets/FosTrackerServlet/GetNumberOfStoresByStatus.java (hunk 4 above)
Comment (Pragya1805) [reply FosServer/src/main/java/com/example/fostracker/servlets/FosTrackerServlet/GetNumberOfStoresByStatus.java]: It did not remove these spaces. But now I've removed them manually

**Review by trblessy — APPROVED** [review]

**Pull request merged**